### PR TITLE
refactor(storage): build rowset while appending data in ColumnMemTable

### LINKED ISSUE #409: storage: build rowset instead of buffering data when appending chunks
Currently, when building an in-memory `ColumnMemTable` rowset,

https://github.com/risinglightdb/risinglight/blob/78caff0ef7937a0b86200cf74c6672eff6e197ee/src/storage/secondary/rowset/mem_rowset.rs#L88-L90

We simply use in-memory array builder to store information, and...

https://github.com/risinglightdb/risinglight/blob/78caff0ef7937a0b86200cf74c6672eff6e197ee/src/storage/secondary/rowset/mem_rowset.rs#L132-L144

We encode them all-at-once at the end.

This would make the ingestion process not-so-efficient. We should be able to encode the rowset when appending chunks into the `ColumnMemTable`, instead of all-at-once.

### PR DESCRIPTION
**Pull request by zzl200012**
Signed-off-by: asuka <312856403@qq.com>

Fixed #409

### CODE REVIEW

**Comment by TennyZhuang** [comment]
Please format your codes with “cargo fly”

File: src/storage/secondary/transaction.rs [hunk 1]
```diff
@@ -97,17 +97,11 @@ impl SecondaryTransaction {
         } else {
             return Ok(());
         };
-        let rowset_id = self.table.generate_rowset_id();
+        let rowset_id = self.table.get_current_rowset_id();
```
Comment (skyzh) [inline src/storage/secondary/transaction.rs:100]: This would cause problems in multi-thread cases. Better to store `rowset_id` in the memtable struct.

**Review by skyzh — COMMENTED** [review]
Rest LGTM, good work!

File: src/storage/secondary/transaction.rs (hunk 1 above)
Comment (zzl200012) [reply src/storage/secondary/transaction.rs]: updated

**Review by skyzh — APPROVED** [review]
LGTM, thanks!

**Pull request merged**